### PR TITLE
feat: more backtrack improvements

### PR DESCRIPTION
**Pull request by larryngton2**
why not

- pauseOnHit, _self-explanatory_
- trackingBuffer, _time to continue to lag for when the target is out of the range you can attack (not the range you specify in the backtrack module_)
- set the max backtrack range to 10 blocks too

### CODE REVIEW

**Comment by larryngton2** [comment]
im not using the chronometer for the tracking buffer because it interferes with the next backtrack delay's chronometer

**Comment by ccetl** [comment]
> im not using the chronometer for the tracking buffer because it interferes with the next backtrack delay's chronometer

Why not create a new chronometer object?

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt [hunk 1]
```diff
@@ -306,13 +313,39 @@ object ModuleBacktrack : Module("Backtrack", Category.COMBAT) {
...
-        target.shouldBeAttacked() &&
-            target.boxedDistanceTo(player) in range &&
+    init {
+        handler<AttackEvent> {
+            val enemy = it.enemy
+            shouldPause = enemy is LivingEntity && enemy.hurtTime < 10
+        }
+    }
+
+    private fun shouldPause(): Boolean {
```
Comment (ghost) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt]:
```
private fun shouldPause(): Boolean {
    return pauseOnHit && shouldPause
}
```

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt [hunk 2]
```diff
@@ -50,17 +51,23 @@ import net.minecraft.util.math.Vec3d
...
     private val chance by float("Chance", 50f, 0f..100f, "%")
+    private val pauseOnHit by boolean("PauseOnHit", false)
     private val espMode = choices("EspMode", Wireframe, arrayOf(Box, Model, Wireframe, None)).apply {
         doNotIncludeAlways()
     }
 
     private val packetQueue = LinkedHashSet<DelayData>()
     private val chronometer = Chronometer()
 
+    private var lastTargetTime = 0L
```
Comment (1zun4) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt]: As @ccetl said: Why not create a new chronometer object?

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt [hunk 3]
```diff
@@ -306,13 +312,35 @@ object ModuleBacktrack : Module("Backtrack", Category.COMBAT) {
...
         enabled && packetQueue.isNotEmpty()
 
-    private fun shouldBacktrack(target: Entity) =
-        target.shouldBeAttacked() &&
-            target.boxedDistanceTo(player) in range &&
+    init {
+        handler<AttackEvent> {
+            val enemy = it.enemy
+            shouldPause = enemy is LivingEntity && enemy.hurtTime < 10
+        }
```
Comment (1zun4) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt]: What on earth? Why is it here? Sure, that works too, but we agreed to use variables to make it look like a function.

Comment (larryngton2) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt]: when i moved it to the existing attack handler it was mostly just returning false, it works on here for some reason

Comment (1zun4) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt]: yeah because there is already an attack handler??

**Pull request merged**